2020-03-Autumn/0_sliding_window/README.md

```diff
@@ -94,21 +105,62 @@ Go to [Solution](#s3)   [Top](#top)
 <!-- Don't remove -->
 <a name="s1"/>
 
-### 1. SOLUTION 1 TODO :bug:
-
-Source: TODO :bug:
-
-#### Naive/Simple Solution
-
-TODO :bug:
+### 1. Max Consecutive Ones
+
+Source: [LeetCode](https://leetcode.com/problems/max-consecutive-ones/)
+
+#### Brute Force Solution
+
+At the first glance of this question, we can see that the program wants to 
+count the most 1s that are grouped together. As we go through the binary 
+array, we can count along the way: If it is a 0, we would stop counting 
+and reset the counter to 0. Otherwise, we would increment the counter by 1. 
+The maximum number of the counter will always be stored in a variable called 
+`max`.
+
+```
+int findMaxConsecutiveOnes(vector<int>& nums) {
+    int max = 0;
+    int count = 0;
+    for (int i = 0; i < nums.size(); ++i) {
+        if (nums[i] == 1) {
+            count++;
+            max = std::max(count, max);
+        } else {
+            count = 0;
+        }
+    }
+    return max;
+}
+```
 
 #### Optimal Solution
 
-TODO :bug:
+Instead of counting the maximum numbers of 1s in a group, we can also define 
+where the group is for faster calculation. In the following solution, the program 
+uses two indecies to keep track of the start and the end of a group of 1s. This is 
+called the "Sliding window" technique. You can move this window around while going 
+through the binary array, checking its value and update the maximum variable.
+
```

Comment:
Maybe mention the time complexity? 
Specifically how it's gone from `O(N^2)` ---> `O(N)`

Reply:
Changes addressed in commit 8a11068.